stingray/events.py

```diff
@@ -246,14 +265,59 @@ def to_lc(self, dt, tstart=None, tseg=None):
         -------
         lc: :class:`stingray.Lightcurve` object
         """
-        if tstart is None and self.gti is not None:
-            tstart = self.gti[0][0]
-            tseg = self.gti[-1][1] - tstart
-
         return Lightcurve.make_lightcurve(
-            self.time, dt, tstart=tstart, gti=self.gti, tseg=tseg, mjdref=self.mjdref
+            self.time, dt, tstart=tstart, gti=self._gti, tseg=tseg, mjdref=self.mjdref
         )
 
+    def to_timeseries(self, dt, array_attrs=None):
+        """Convert the event list to a :class:`stingray.StingrayTimeseries` object.
+
+        The result will be something similar to a light curve, but with arbitrary
+        attributes corresponding to a weighted sum of each specified attribute of
+        the event list.
+
+        E.g. if the event list has a ``q`` attribute, the final time series will
+        have a ``q`` attribute, which is the sum of all ``q`` values in each time bin.
```

Comment:
Yeah, I still don't understand this. :D 
Does this mean that if you have a `q` that's multi-dimensional, say 3 values for every time bin, the resulting `TimeSeries` will have the sum of these three values for each bin in its attribute `q`? Why? Also, above it says "weighted sum". Weighted by what?

Reply:
Ooooh, I see. It gets *binned*. But ... why? I thought `TimeSeries` could accommodate EventLists? Wasn't that the whole point of making it more general?

Reply:
I think we need to think a bit carefully here how to structure this. Because right now, there are two separate behaviours supported by the same class: (1) You could make a TimeSeries object and just stick events into it, and that would work fine, or (2) you could generate a `TimeSeries` out of an EventList, but then it gets binned, so it's more like `Lightcurve`. I don't mind (2) intrinsically, but then we should not expose the user to the possibility of (1) even if we do that under the hood. I think I'd rather generalise `EventList` into something more general with more attributes if necessary, rather than allow the user to store EventLists as a time series object.

Reply:
EventList is clearly a special case of a generic time series, and it makes sense to inherit much of the `StingrayTimeseries` class. On the other hand, creating a binned time series from an event list is useful.
I renamed the method as `to_binned_timeseries`, I think this is the easiest way to solve the conundrum. I have a time series object, I transform it into some other kind of time series object

Reply:
Yeah, I think that's much better!